### PR TITLE
Remove deprecated api v2

### PR DESCRIPTION
**Pull request by Automaat**

### CODE REVIEW

**Review by Ferdudas97 — DISMISSED** [review]

**Comment by Automaat** [comment]
I've tested this change on our Allegro environments. It all looks fine. We could merge it already. I am waiting for review from outside of our organisation. @rulex123 Could you look at it?

File: cache/src/main/java/io/envoyproxy/controlplane/cache/Resources.java
Comment (rulex123) [inline cache/src/main/java/io/envoyproxy/controlplane/cache/Resources.java:221]: Can you update the javadoc?

File: cache/src/main/java/io/envoyproxy/controlplane/cache/Resources.java
Comment (rulex123) [inline cache/src/main/java/io/envoyproxy/controlplane/cache/Resources.java:34]: There's a static inner class called `V2` that perhaps cal also be removed?

File: README.md
Comment (rulex123) [inline README.md:10]: I think we should drop any references to migrating to v2 in the readmes, since with this patch we are effectively removing the code that made those migration paths possible.

File: server/src/test/java/io/envoyproxy/controlplane/server/serializer/DefaultProtoResourcesSerializerTest.java
Comment (rulex123) [inline server/src/test/java/io/envoyproxy/controlplane/server/serializer/DefaultProtoResourcesSerializerTest.java:24]: Do you think we could remove the tests for v2 resources altogether?

File: server/src/test/java/io/envoyproxy/controlplane/server/serializer/CachedProtoResourcesSerializerTest.java
Comment (rulex123) [inline server/src/test/java/io/envoyproxy/controlplane/server/serializer/CachedProtoResourcesSerializerTest.java:24]: Same question as for the other serializer class.

File: cache/src/main/java/io/envoyproxy/controlplane/cache/Resources.java
Comment (Automaat) [reply cache/src/main/java/io/envoyproxy/controlplane/cache/Resources.java:65]: Hmm, I was using Google Java style format as specified [here](https://github.com/envoyproxy/java-control-plane/blob/main/STYLE.md)

File: cache/src/main/java/io/envoyproxy/controlplane/cache/XdsRequest.java
Comment (Automaat) [reply cache/src/main/java/io/envoyproxy/controlplane/cache/XdsRequest.java]: fixed

File: .circleci/config.yml
Comment (Automaat) [reply .circleci/config.yml]: removed

File: cache/src/main/java/io/envoyproxy/controlplane/cache/Resources.java
Comment (Automaat) [reply cache/src/main/java/io/envoyproxy/controlplane/cache/Resources.java]: updated

File: cache/src/main/java/io/envoyproxy/controlplane/cache/Resources.java
Comment (Automaat) [reply cache/src/main/java/io/envoyproxy/controlplane/cache/Resources.java:34]: I must have missed it in the first pass, removed now

File: README.md
Comment (Automaat) [reply README.md:10]: There is link to documentation one in the next line. I've fixed link right now

File: server/src/test/java/io/envoyproxy/controlplane/server/serializer/CachedProtoResourcesSerializerTest.java
Comment (Automaat) [reply server/src/test/java/io/envoyproxy/controlplane/server/serializer/CachedProtoResourcesSerializerTest.java:24]: removed

File: api/pom.xml
Comment (rulex123) [reply api/pom.xml:8]: ping @Automaat

File: cache/src/main/java/io/envoyproxy/controlplane/cache/Resources.java
Comment (rulex123) [reply cache/src/main/java/io/envoyproxy/controlplane/cache/Resources.java:223]: ping @Automaat

**Comment by rulex123** [comment]
LGTM in general, but there are still a couple of open questions if you could PTAL (I pinged you directly on the specific comments)

File: .circleci/config.yml
Comment (Automaat) [reply .circleci/config.yml]: removed

File: cache/src/main/java/io/envoyproxy/controlplane/cache/Resources.java
Comment (Automaat) [reply cache/src/main/java/io/envoyproxy/controlplane/cache/Resources.java:223]: good catch, I removed it by mistake

Comment (Automaat) [reply cache/src/main/java/io/envoyproxy/controlplane/cache/Resources.java:223]: reverted this change

File: server/src/test/java/io/envoyproxy/controlplane/server/TestMain.java
Comment (Automaat) [reply server/src/test/java/io/envoyproxy/controlplane/server/TestMain.java]: Changed how `TestResources.createCluster` method works. Now it includes you can specify the `DiscoveryType`

File: api/pom.xml
Comment (Automaat) [reply api/pom.xml]: done

**Pull request merged**
